### PR TITLE
feat: move freezing logic in native-stack to navigation

### PR DESCRIPTION
**Pull request by WoLewicki**
**Motivation**

PR connected to https://github.com/software-mansion/react-native-screens/pull/2566 moving the freezing logic to `react-navigation` since we have control over `preloaded` screens here.

**Test plan**

`TestFreeze` in `RNScreens` repo.

### CODE REVIEW

File: packages/native-stack/src/views/NativeStackView.native.tsx [hunk 1]
```diff
@@ -500,11 +509,16 @@ export function NativeStackView({
             preloadedDescriptors[route.key] !== undefined &&
             descriptors[route.key] === undefined;
 
+          const shouldFreeze = isFabric()
+            ? !isPreloaded && !isFocused && !isBelowFocused
```
Comment (satya164) [inline packages/native-stack/src/views/NativeStackView.native.tsx:513]: Interesting why we can't freeze the previous screen on Fabric. Maybe we can add a comment saying why?

**Comment by WoLewicki** [comment]
Ok I updated PR in RNScreens and added same logic for rest of navigators. Would be nice if it was tested in all of them.

**Review by satya164 — APPROVED** [review]
It seems to work so I'm merging this

File: packages/native-stack/src/views/NativeStackView.native.tsx (hunk 1 above)
Comment (hirbod) [reply packages/native-stack/src/views/NativeStackView.native.tsx:514]: Maybe we should do the same for Paper as well? I noticed that many of NativeWind's active classes are also frozen while navigating. Watch the video, where we scale the image on press and it stays scaled until we return. The effect is cool sometimes, as it reminds me of cupertino lists keep the active item active until you return, but it's sometimes a bit weird too. 

https://github.com/user-attachments/assets/45e595c0-0b9a-412e-a619-a8ea72952357

and another example, what the Deposit button.

https://github.com/user-attachments/assets/c1c690cb-7a3f-43a9-9ef4-5c682b1a8a7a

Comment (satya164) [reply packages/native-stack/src/views/NativeStackView.native.tsx:514]: I think the way it's currently described (freeze inactive screens), it's strange not freeze an inactive screen. Maybe this freeze feature needs a rethinking - though I think that's out of the scope of this PR.

Comment (hirbod) [reply packages/native-stack/src/views/NativeStackView.native.tsx:514]: It is out of the scope, true, but that line reminded me of "this issue" and I just wanted to share it with you

Comment (hirbod) [reply packages/native-stack/src/views/NativeStackView.native.tsx:514]: I've created a [discussion for it ](https://github.com/react-navigation/react-navigation/discussions/12337)

**Pull request merged**